Cubical/HITs/Sn/Properties.agda

```diff
@@ -13,29 +12,329 @@ open import Cubical.Foundations.Prelude
+                                                    (f y))) ] _
+  indStep = wedgeConSn n m (λ _ _ → isOfHLevelPathP' (suc (n + suc m)) (hlev _ _) _ _)
+                           (λ a i → transp (λ i₁ → A (merid (ptSn (suc n)) (i₁ ∧ i)) a) (~ i)
+                                            (f a))
+                           guy
+                           guyId
+
+  F : (x : Susp (S₊ (suc n))) (y : S₊ (suc m))  → A x y
+  right : (x : Susp (S₊ (suc n))) → F x (ptSn (suc m)) ≡ g x
+  F north y = f y
+  F south y = subst (λ x → A x y) (merid (ptSn (suc n))) (f y)
+  F (merid a i) y = indStep .fst a y i
+  right north = sym hom
+  right south = cong (subst (λ x → A x (ptSn (suc m)))
+                            (merid (ptSn (suc n))))
+                            (sym hom)
+              ∙ λ i → transp (λ j → A (merid (ptSn (suc n)) (i ∨ j)) (ptSn (suc m))) i
+                              (g (merid (ptSn (suc n)) i))
+  right (merid a i) j =
+    hcomp (λ k → λ { (i = i0) → hom (~ j)
+                    ; (i = i1) → transpLemma (ptSn (suc n)) j
+                    ; (j = i0) → indStep .snd .snd a (~ k) i
+                    ; (j = i1) → g (merid a i)})
+          (hcomp (λ k →  λ { (i = i0) → hom (~ j)
+                            ; (i = i1) → compPath-lem (transpLemma a) (transpLemma (ptSn (suc n))) k j
+                            ; (j = i1) → g (merid a i)})
+                 (hcomp (λ k → λ { (i = i0) → hom (~ j)
+                                  ; (j = i0) → transp (λ i₂ → A (merid a (i₂ ∧ i)) (ptSn (suc m))) (~ i)
+                                                       (f (ptSn (suc m)))
+                                  ; (j = i1) → transp (λ j → A (merid a (i ∧ (j ∨ k))) (ptSn (suc m))) (k ∨ ~ i)
+                                                       (g (merid a (i ∧ k))) })
+                        (transp (λ i₂ → A (merid a (i₂ ∧ i)) (ptSn (suc m))) (~ i)
+                                (hom (~ j)))))
+
+-- We get ∥ Sⁿ⁺² ∥ₙ₊₂ ≃ ∥ Ω Sⁿ⁺³ ∥ₙ₊₂ from the Freudenthal suspension theorem.
+-- Cavallos proof of the Freudenthal suspenion theorem (see Cubical.Homotopy.Freudenthal)
+-- can be modified to completely avoid theory about connectedness if one
+-- only considers (n+2)-spheres:
+
+module miniFreudenthal (n : HLevel) where
+  σ : S₊ (2 + n) → typ (Ω (S₊∙ (3 + n)))
+  σ a = merid a ∙ merid north ⁻¹
+
+  S2+n = S₊ (2 + n)
+  4n+2 = (2 + n) + (2 + n)
+
+  module WC-S (p : north ≡ north) where
+    P : (a b : S₊ (2 + n)) → Type₀
+    P a b = σ b ≡ p → hLevelTrunc 4n+2 (fiber (λ x → merid x ∙ merid a ⁻¹) p)
+
+    hLevelP : (a b : S₊ (2 + n)) → isOfHLevel 4n+2 (P a b)
+    hLevelP _ _ = isOfHLevelΠ 4n+2 λ _ → isOfHLevelTrunc 4n+2
+
+    leftFun : (a : S₊ (2 + n)) → P a north
+    leftFun a r = ∣ a , (rCancel' (merid a) ∙ rCancel' (merid north) ⁻¹) ∙ r ∣
+
+    rightFun : (b : S₊ (2 + n)) → P north b
+    rightFun b r = ∣ b , r ∣
+
+    funsAgree : leftFun north ≡ rightFun north
```

Comment:
Wouldn't this proof be easier to read if you just wrote it directly using dimension variables?

Reply:
Sorry, maybe I'm a bit slow, but I don't understand hehe. Do you mean doing the whole of "miniFreudenthal" directly?

Reply:
No, I just meant in `funsAgree`. I was thinking that it might be easier to read if you inline `funExt` and `cong'`, but I might be wrong

Reply:
Oh, to be honest I barely looked at this. This is Evan's stuff. I'm guessing he had a reason for using the obscure `cong`' hehe.

Reply:
Oh but yeah, I agree now. Fixed this. Realised the original proof looked slightly different in general.

Reply:
I was told to! :P https://github.com/agda/cubical/pull/275#discussion_r406647042